localstack/services/dynamodb/provider.py

```diff
@@ -1739,8 +1722,10 @@ def get_updated_records(table_name: str, existing_items: List) -> List:
           into the PartiQL query execution inside DynamoDB Local and directly extract the list of updated items.
     """
     result = []
+    # TODO: remove get_region(), but the logic is all around dynamodb
+    region = aws_stack.get_region()
     stream_spec = dynamodb_get_table_stream_specification(
-        account_id=get_aws_account_id(), region_name=aws_stack.get_region(), table_name=table_name
+        account_id=get_aws_account_id(), region_name=region, table_name=table_name
     )
```

Comment:
With https://github.com/localstack/localstack/pull/9382 merged, you should use `account_id` and `region_name` available as function args

Reply:
of course yes, this was about the `get_updated_records` method, but I've updated the signature and passed the context account and region instead, as it's used in only one place so not a lot of refactoring. Good luck for dynamodb, it is really everywhere there 😬

Reply:
It's now rebased so it properly uses the context! Quite the rebasing but it does feel better 😄